reactioncommerce/api-service-catalogs
# fix: ID decoding

Resolves #3
Impact: **critical**
Type: **bugfix**

## Issue
`slugOrId` decoding was broken due to an update to `api-utils`

## Solution

- Pin to the latest version of `api-utils`
- Update decoding logic in resolver and Mongo query

## Breaking changes

none

## Testing
1. check out repo next to your `reaction` repo (Reaction 3.8.0 branch: https://github.com/reactioncommerce/reaction/pull/6253)
2. from your `reaction` directory run `bin/package-link api-plugin-catalogs ../api-plugin-catalogs`
3. Run the queries

```
# Get shop id
query q1 {
  primaryShopId  
}

# Get products
query q2 {
	catalogItems(shopIds: ["<SHOP_ID>"]) {
    nodes {
      _id
      ... on CatalogItemProduct {
        product {
          slug
        }
      }
    }
  }
}

# query by the slug
query q3 {
  catalogItemProduct(
    shopId: "<SHOP_ID>", 
    slugOrId: "<SLUG>"
  ) {
    _id
    product {
      slug
    }
  }
}

#query by the catalog item ID (not the product id)
query q4 {
  catalogItemProduct(
    shopId: "<SHOP_ID>", 
    slugOrId: "<CATALOG_ITEM_ID>"
  ) {
    _id
    product {
      slug
    }
  }
}
```
